scanpy/tools/_louvain.py

```diff
@@ -106,6 +106,9 @@ def louvain(
             restrict_categories,
             adjacency,
         )
+    uns_key = 'louvain' if key_added == 'louvain' else f'louvain_{key_added}'
```

Comment:
You could also catch the case if someone puts `key_added='louvain_newkey'`.
`uns_key = key_added if key_added.startswith('louvain') else f'louvain_{key_added}'`

Reply:
Sounds good! I didn't want to overengineer it but people use louvain/leiden prefix a lot, so I did it :)

Reply:
I am "people" in this case ;).